slackbot.go

```diff
@@ -0,0 +1,189 @@
+	bot.Command("open emergency <msg>",
+		"Open an EMERGENCY incident to Customer Support",
+		func(request slacker.Request, response slacker.ResponseWriter) {
+			Emergency(request, response, config)
+		})
+
+	// when no other "Command" matches and text is sent to the Bot, this function will be run instead
+	bot.DefaultCommand(
+		func(request slacker.Request, response slacker.ResponseWriter) {
+			Default(request, response, config)
+		})
+
+	// function run for all events received by the bot (including time ticks)
+	bot.DefaultEvent(
+		func(event interface{}) {
+			//log.Println(event)
+		})
+
+	// set the "help" message handling function
+	bot.Help("help",
+		slacker.WithHandler(
+			func(request slacker.Request, response slacker.ResponseWriter) {
+				help(response, config)
+			}))
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	err := bot.Listen(ctx)
+	if err != nil {
+		Err(err.Error())
+	}
+
+	return nil
+}
+
+// Init is called upon Bot creation (first startup)
+func Init(config *config) {
+	fmt.Printf("Connected!\n")
+	client := pagerduty.NewClient(config.pagerDuty.aPIKey)
+	var opts pagerduty.GetScheduleOptions
+
+	schedule, err := client.GetSchedule(config.pagerDuty.schedule, opts)
+	if err != nil {
+		trace.Wrap(err)
+	}
+	fmt.Printf("Configured schedule is \"%s\" with ID: %s\n", schedule.Name,
+		config.pagerDuty.schedule)
+}
+
+// Err function is used to handle all Error reported by the Bot
+func Err(err string) {
+	fmt.Fprint(os.Stderr, err)
+}
+
+// Emergency is used to open Emergency Incidents on PagerDuty
+func Emergency(request slacker.Request, response slacker.ResponseWriter, config *config) {
+	client := pagerduty.NewClient(config.pagerDuty.aPIKey)
+	var scheduleOpts pagerduty.GetScheduleOptions
+
+	schedule, err := client.GetSchedule(config.pagerDuty.schedule, scheduleOpts)
+	if err != nil {
+		textErr := fmt.Sprintf("Error encountered while fetching schedules: %s", err.Error())
+		response.Reply(textErr)
+		trace.Wrap(err)
+	}
+	fmt.Printf(`Opening incident on schedule "%s"/%s`, schedule.Name,
+		config.pagerDuty.schedule)
+
+	newIncident := pagerduty.CreateIncidentOptions{
+		Type: "incident",
+		Title: fmt.Sprintf("Incident opened by %s, via Slack/@%s",
+			config.customerName, config.slack.botUsername),
+	}
+
+	newIncident.Body = pagerduty.APIDetails{
+		Type:    "incident_body",
+		Details: request.Param("msg"),
+	}
+
+	newIncident.Service = pagerduty.APIReference{
+		Type: "service_reference",
+		ID:   config.pagerDuty.service,
+	}
+
+	createIncidentOpts := pagerduty.CreateIncident{
+		Incident: newIncident,
+	}
+
+	incident, err := client.CreateIncident(config.pagerDuty.fromEmail, &createIncidentOpts)
+	if err != nil {
+		errText := "There was an error while creating a new incident created, please try again and report the following error" + err.Error()
+		Err(errText)
+		response.Reply(errText)
```

Comment:
You should return here. Otherwise you're also sending the successful response

Reply:
is it okay now?

Reply:
https://godoc.org/github.com/shomali11/slacker#ResponseWriter
response.ReportError(err) instead of Reply

Reply:
fixed.